### DISCUSSION TITLE
How to request user:email scope with GitHub Apps?

### DISCUSSION THREAD

**Discussion by yykamei** [community/community#22913] [New to GitHub]
<p>Hi there,</p>
<p>I’m developing a GitHub App that requires User-to-server requests.</p>
<p>As well as OAuth Apps, GitHub Apps can identify users via almost same its flow, but it cannot be set <strong>scope</strong> parameter. Therefore I cannot get emails of users because listing user emails requires <strong>user:email</strong>  scope.<br>
<a href="https://developer.github.com/v3/users/emails/#list-email-addresses-for-a-user" rel="nofollow noopener">This API</a> is not listed in User-to-server requests <a href="https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/#users" rel="nofollow noopener">supported endpoints</a>.</p>
<p>Does GitHub have a roadmap to support to set a  <strong>scope</strong> for GitHub Apps?</p>
<p>Or are there any other solutions?</p>
<p>Thanks,</p>

**Comment by lee-dohm** [comment]
<p>It’s my understanding that GitHub Apps are intended to perform actions on a user’s or organization’s repositories. Even in the case of a user-to-server request, it is still performing actions in relation to the installation’s repositories. So what action on a repository are you attempting to perform that requires the user’s email address?</p>

**Comment by yykamei** [comment]
<blockquote>
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> wrote:</p>
</blockquote>
<blockquote>
<p>It’s my understanding that GitHub Apps are intended to perform actions on a user’s or organization’s repositories. Even in the case of a user-to-server request, it is still performing actions in relation to the installation’s repositories. So what action on a repository are you attempting to perform that requires the user’s email address?</p>
</blockquote>
<p>Actually, we don’t need user’s email addresses for actions on repositories. However, there are no ways to notify users of errors in our service without their email addresses. Of course, we can notify users via Issues, Pull Requests or other GitHub related UI, but it’s not appropriate for something not related to repositories.</p>
<p>In this case, should we have both GitHub Apps and OAuth Apps? In other words, should a user-to-server request be used for only repository related actions?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>It is my understanding that user-to-server requests are designed only for repository-related actions, yes.</p>

**Comment by yykamei** [comment]
<p>I understood. Thank you for your replay.</p>
